cmd/osv-scanner/fix/main.go

```diff
@@ -0,0 +1,220 @@
+	autoModeCategory = "non-interactive options:" // intentionally lowercase to force it to sort after the other categories
+)
+
+type osvFixOptions struct {
+	remediation.RemediationOptions
+	Client     client.ResolutionClient
+	Manifest   string
+	ManifestRW manifest.ManifestIO
+	Lockfile   string
+	LockfileRW lockfile.LockfileIO
+	RelockCmd  string
+}
+
+func Command(stdout, stderr io.Writer, r *reporter.Reporter) *cli.Command {
+	return &cli.Command{
+		Name:        "fix",
+		Usage:       "[EXPERIMENTAL] scans a manifest and/or lockfile for vulnerabilities and suggests changes for remediating them",
+		Description: "[EXPERIMENTAL] scans a manifest and/or lockfile for vulnerabilities and suggests changes for remediating them",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:      "manifest",
+				Aliases:   []string{"M"},
+				Usage:     "manifest file to remediate vulnerabilities in",
+				TakesFile: true,
+			},
+			&cli.StringFlag{
+				Name:      "lockfile",
+				Aliases:   []string{"L"},
+				Usage:     "lockfile to remediate vulnerabilities in",
+				TakesFile: true,
+			},
+			&cli.StringFlag{
+				Name:  "data-source",
+				Usage: "source to fetch package information from; value can be: deps.dev, native",
+				Value: "deps.dev",
+				Action: func(ctx *cli.Context, s string) error {
+					if s != "deps.dev" && s != "native" {
+						return fmt.Errorf("unsupported data-source \"%s\" - must be one of: deps.dev, native", s)
+					}
+
+					return nil
+				},
+			},
+			&cli.StringFlag{
+				Name:  "relock-cmd",
+				Usage: "command to run to regenerate lockfile on disk after changing the manifest",
+			},
+
+			&cli.BoolFlag{
+				Name:  "non-interactive",
+				Usage: "run in the non-interactive mode",
+				Value: !term.IsTerminal(int(os.Stdin.Fd())), // Default to non-interactive if not being run in a terminal
+			},
+			&cli.StringFlag{
+				Category: autoModeCategory,
+				Name:     "strategy",
+				Usage:    "remediation approach to use; value can be: in-place, relock",
+				Value:    "relock",
+				Action: func(ctx *cli.Context, s string) error {
+					if !ctx.Bool("non-interactive") {
+						// This flag isn't used in interactive mode
+						return nil
+					}
+					switch s {
+					case "in-place":
+						if !ctx.IsSet("lockfile") {
+							return fmt.Errorf("in-place strategy requires lockfile")
+						}
+					case "relock":
+						if !ctx.IsSet("manifest") {
+							return fmt.Errorf("relock strategy requires manifest file")
+						}
+					default:
+						return fmt.Errorf("unsupported strategy \"%s\" - must be one of: in-place, relock", s)
+					}
+
+					return nil
+				},
+			},
+			&cli.IntFlag{
+				Category: autoModeCategory,
+				Name:     "apply-top",
+				Usage:    "apply the top N patches",
+				Value:    -1,
+			},
+
+			&cli.BoolFlag{
+				Category: upgradeCategory,
+				Name:     "disallow-major-upgrades",
+				Usage:    "disallow major version changes to dependencies",
+			},
+			&cli.StringSliceFlag{
```

Comment:
I think we discussed this, but we need to think of a more extensible semantics for this.

e.g. something like 

```
--disallow-upgrades=PKG-1:major,PKG-2, 
```

Can be a TODO for now.

Reply:
Added a TODO